src/mlpack/tests/main_tests/sparse_coding_test.cpp

```diff
@@ -0,0 +1,465 @@
+{
+  mat inputData;
+  inputData.load("mnist_first250_training_4s_and_9s.arm");
+
+  // Test for L1 value.
+
+  // Input training data.
+  SetInputParam("training", inputData);
+  SetInputParam("atoms", (int) 10);
+  SetInputParam("lambda1", (double) -1.0);
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+
+  // Test for L2 value.
+
+  // Input training data.
+  SetInputParam("training", inputData);
+  SetInputParam("atoms", (int) 10);
+  SetInputParam("lambda2", (double) -1.0);
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+
+  // Test for max_iterations.
+
+  // Input training data.
+  SetInputParam("training", inputData);
+  SetInputParam("atoms", (int) 10);
+  SetInputParam("max_iterations", (int) -1.0);
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+
+  // Test for objective_tolerance.
+
+  // Input training data.
+  SetInputParam("training", inputData);
+  SetInputParam("atoms", (int) 10);
+  SetInputParam("objective_tolerance", (double) -1.0);
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+
+  // Test for newton_tolerance.
+
+  // Input training data.
+  SetInputParam("training", inputData);
+  SetInputParam("atoms", (int) 10);
+  SetInputParam("newton_tolerance", (double) -1.0);
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+
+  // Test for atoms.
+
+  // Input training data.
+  SetInputParam("training", std::move(inputData));
+  SetInputParam("atoms", (int) 0);
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Make sure atoms are specified if training data is passed.
+ */
+BOOST_AUTO_TEST_CASE(SparseCodingReqAtomsTest)
+{
+  mat inputData;
+  inputData.load("mnist_first250_training_4s_and_9s.arm");
+
+  // Input training data.
+  SetInputParam("training", std::move(inputData));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Ensure only one of input_model or initial_dictionary
+ * is specified.
+ */
+BOOST_AUTO_TEST_CASE(SparseCodingModelVerTest)
+{
+  mat inputData;
+  inputData.load("mnist_first250_training_4s_and_9s.arm");
+
+  // Shuffle input dataset.
+  inputData = shuffle(inputData);
+
+  // Input data.
+  SetInputParam("training", std::move(inputData));
+  SetInputParam("atoms", (int) 30);
+  SetInputParam("max_iterations", (int) 10);
+  SetInputParam("normalize", (bool) true);
+
+  mlpackMain();
+
+  mat initialDictionary;
+  initialDictionary = std::move(CLI::GetParam<arma::mat>
+                                    ("dictionary"));
```

Comment:
This is a really minor comment, but you could initialize and assign on the same line, and this might make the overflow a little easier:

```
mat initialDictionary = 
    std::move(CLI::GetParam<arma::mat>("dictionary"));
```

Reply:
Sure, will reformat it everywhere in PR.